src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -289,22 +288,27 @@ private StripeError parseStripeError(
   }
 
   private void handleError(StripeResponse response, ApiMode apiMode) throws StripeException {
-    JsonObject responseBody = ApiResource.GSON.fromJson(response.body(), JsonObject.class);
-
-    /*
-     OAuth errors are JSON objects where `error` is a string. In
-     contrast, in API errors, `error` is a hash with sub-keys. We use
-     this property to distinguish between OAuth and API errors.
-    */
-    if (responseBody.has("error") && responseBody.get("error").isJsonPrimitive()) {
-      JsonPrimitive error = responseBody.getAsJsonPrimitive("error");
-      if (error.isString()) {
-        handleOAuthError(response);
+    try {
+      /*
+      OAuth errors are JSON objects where `error` is a string. In
+      contrast, in API errors, `error` is a hash with sub-keys. We use
+      this property to distinguish between OAuth and API errors.
+
+      Try to read the response body to see if it must be
+      */
+      JsonObject responseBody = ApiResource.GSON.fromJson(response.body(), JsonObject.class);
+      if (responseBody.has("error") && responseBody.get("error").isJsonPrimitive()) {
+        JsonPrimitive error = responseBody.getAsJsonPrimitive("error");
+        if (error.isString()) {
+          handleOAuthError(response);
+        }
+      } else if (apiMode == ApiMode.V2) {
+        handleV2ApiError(response);
+      } else {
+        handleV1ApiError(response);
       }
-    } else if (apiMode == ApiMode.V2) {
-      handleV2ApiError(response);
-    } else {
-      handleV1ApiError(response);
+    } catch (JsonSyntaxException e) {
```

Comment:
Not directly related to the issue but are there any other exceptions that can be thrown here that we want to wrap in StripeException?

Reply:
It only expects to throw `JsonSyntaxException` https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/JsonSyntaxException.html. Should we wrap for all unexpected exceptions as well?

Reply:
I'm not sure what the norm for our SDKs is here.  I think if we don't have a concrete answer now, maybe we leave it for later.